lib/commons/color/get-background-color.js

```diff
@@ -188,7 +188,9 @@ function sortPageBackground(elmStack) {
  * @return {Array}
  */
 color.getBackgroundStack = function(elm) {
-	let rect = elm.getBoundingClientRect();
+	// allows inline elements spanning multiple lines to be evaluated
+	let multipleRects = elm.getClientRects();
+	let rect = multipleRects.length > 1 ? multipleRects[0] : elm.getBoundingClientRect();
```

Comment:
I found that if I used `getClientRects()` to gather an x, y to pass to document.elementsFromPoint, we could leverage that method to find elements on top. The only downside I could see is if a point in rects[1+] didn't overlap, the background stack could be different for those nodes.  A scenario could be if an inline element spanning multiple lines is only partially covered–we could potentially put it into incomplete if there's more than one clientRect but we'd have to check if something with color overlaps partially for that result to be helpful, and that seems like super overkill at this time.

Reply:
I think there's a possible false positive we're ignoring if we're not checking each of those clientRects. I don't think we should put them all in incomplete. Instead I'd suggest we do elementsFromPoint on each of them, and only run it into incomplete if the resulting arrays don't match. WDYT?

Reply:
@WilcoFiers I found one wrinkle in this plan: if the inline element has a line break in it (a la `<br>`), getClientRects() includes an extra rect for it. Perhaps I need to exclude that case, since it doesn't add anything relevant in terms of text content impacted by color? It would be a shame for axe-core to get hung up on a line break.

Reply:
I also need a method for comparing arrays of DOM objects, if you have any performant suggestions.

Reply:
No preference. Shallow compare is pretty easy though:

```js
arr1.length == arr2.length && arr1.every((val, i) => val === arr2[i])
```

I wouldn't try to optimise this. This isn't nearly as costly as many other things the color-contrast rule is doing.

Reply:
Ha, that's really similar to what I ended up doing. But I still have to do a lot of finagling with the data structure. I'm getting close though.